docs/DeveloperGuide.md

```diff
@@ -162,40 +162,162 @@ Classes used by multiple components are in the `seedu.realodex.commons` package.
 
-The Filter by Name feature allows users to filter the list of persons in Realodex based on their names. 
-It is implemented using a predicate that checks if a person's name contains the keyphrase provided by the user. 
-The key components involved in this feature include:
+- `FilterCommand`: Executes the filtering operation based on a provided predicate that encapsulates the filtering criteria.
+- `PrefixChecker`: Ensures the syntactic correctness of user inputs by verifying command prefixes. 
+Detects command format violations, and facilitates clear error messaging.
+- `FilterCommandParser`: Parses user input into a FilterCommand by identifying the filtering field and keyphrase. 
+- `PredicateProducer`: Generates specific predicates based on the identified field and keyphrase. 
+- `Predicates`: `NameContainsKeyphrasePredicate`, `RemarkContainsKeyphrasePredicate`, `TagsMatchPredicate`, `BirthdayIsInMonthPredicate`, and `HousingTypeMatchPredicate` 
+that determine if a person's attributes match the user-defined criteria.
+
+#### Filter Command Architecture
+<puml src="diagrams/filter/FilterFeatureArchitecture.puml" width="1000" />
+
+### Filter by Name feature
+
+#### Description
 
-- FilterCommand: A command that, when executed, updates the filtered person list based on the predicate.
-- NameContainsKeyphrasePredicate: A predicate that tests whether a person's name contains the given keyphrase.
+The Filter by Name feature allows users to filter the list of persons in Realodex based on their names. 
+This is implemented using the `NameContainsKeyphrasePredicate` that checks if a person's name contains the keyphrase provided by the user. 
 
 #### Example Usage Scenario
-1. The user launches the application and the Model is initialized with the initial list of persons. 
-2. The user executes the command filter n/John, intending to filter out persons whose names contain "John".
-3. `LogicManager` parses the command, creating a FilterCommand with a NameContainsKeyphrasePredicate initialized with "John".
-4. `FilterCommand` is executed, using the predicate to filter the list of persons.
-5. The UI is updated to show only the persons whose names contain "John".
+1. The user executes the command `filter n/John`, intending to filter out persons whose names contain "John". 
+2. The FilterCommandParser interprets the input, creating a FilterCommand with the `NameContainsKeyphrasePredicate`. 
+3. The `FilterCommand` applies the `NameContainsKeyphrasePredicate` predicate, updating the filtered person list to only include those whose names contain "John".
+4. The UI reflects this filtered list.
 
 #### Design considerations
 
 Aspect: Handling of partial names
 
-**Alternative 1 (current choice): Allow partial matches of names. For example, filter n/Jo will match "John", "Joanna", etc.**
+**Alternative 1 (current choice): Allow partial matches of names.**
+
+> For example, `filter n/Jo` returns persons with names "John", "Joanna", etc.
```

Comment:
In this example, consider including an example that has "jo" in the middle of the name to demonstrate the substring properties (like banjo)

Reply:
Agreed

Reply:
great suggestion!